Refresh.GameServer/Database/GameDatabaseContext.Relations.cs

```diff
@@ -192,6 +192,81 @@ public void ClearQueue(GameUser user)
 
     #region Rating and Reviewing
 
+    public void RateReview(GameReview review, RatingType rt, GameUser user)
+    {
+        // if rating type is neutral, remove previous review rating by user 
+        if (rt == RatingType.Neutral && this.ReviewRatingExistsByUser(user, review))
+        {
+            RateReviewRelation rel =
+                this.RateReviewRelations.First(relation => relation.Review == review && relation.User == user);
+            this.Write(() => this.RateReviewRelations.Remove(rel));
+
+            return;
+        }
+        // swap rating
+        if (this.ReviewRatingExistsByUser(user, review))
+        {
+            RateReviewRelation rel = this.RateReviewRelations.First(relation => relation.Review == review && relation.User == user);
+            
+            this.Write(() => rel.RatingType = rt);
+
+            return;
+        }
+        
+        RateReviewRelation reviewRelation = new()
+        {
+            RatingType = rt,
+            Review = review,
+            User = user,
+        };
+
+        this.Write(() =>
+        {
+            this.RateReviewRelations.Add(reviewRelation);
+        });
+    }
+    
+    public ReviewRating GetRatingForReview(GameReview review)
+    {
+        IQueryable<RateReviewRelation> relations = this.RateReviewRelations.Where((relation => relation.Review == review));
+        ReviewRating rating = new();
+
+        foreach (RateReviewRelation rel in relations)
+        {
+            if (rel.RatingType == RatingType.Yay)
+            {
+                rating.PostiveRating++;
+            }
+            else
+            {
+                rating.NegativeRating++;
+            }
+        }
+
+        return rating;
+    }
+
+    public GameReview GetReviewByUsernameForLevelId(string userName, int levelId)
+    {
+        GameLevel level = this.GameLevels.First(gameLevel => gameLevel.LevelId == levelId);
+        GameUser user = this.GameUsers.First(gameUser => gameUser.Username == userName);
+        
+        GameReview review = this.GameReviews.First((gameReview => gameReview.Publisher == user && gameReview.Level == level));
+        
+        return review;
+    }
+
+    public bool ReviewRatingExistsByUser(GameUser user, GameReview review)
+    {
+        return this.RateReviewRelations.Any(relation => relation.Review == review && relation.User == user);
+    }
+    
+    public bool ReviewRatingExists(GameUser user, GameReview review, RatingType newRating)
+    {
+        return this.RateReviewRelations.Any(relation => relation.Review == review && relation.User == user && relation._ReviewRatingType == (int) newRating);
+    }
```

Comment:
What's the use for this?

Reply:
Ah, the different naming of the function and the naming of the `newRating` parameter confused me.

I think it's okay if this is an overload of `ReviewRatingExistsByUser`. But definitely change `newRating` to just `rating`.